pkg/services/meta/containers.go

```diff
@@ -591,3 +593,121 @@ func (s *containerStorage) handleNewEpoch(e uint64) error {
 
 	return errors.Join(mptErr, dbErr)
 }
+
+func expandChildren(st storage.Store, rootOIDs []byte) []byte {
+	if len(rootOIDs) == 0 {
+		return nil
+	}
+
+	// sorting before every SEEK is used for exact single operation for every
+	// children searching subroutine; search is done in 3 steps:
+	//  1. root -> last/link object search
+	//  2. last/link -> first part ID search
+	//  3. first part ID -> all children search
+
+	var childrenWithParentID [][]byte
+	var rng storage.SeekRange
+	rng.Prefix = slices.Concat([]byte{attrPlainToOIDIndex}, []byte(objectsdk.FilterParentID), object.AttributeDelimiter)
+	rng.Start = make([]byte, oid.Size)
+
+	rootsSorted := slices.SortedFunc(slices.Chunk(rootOIDs, oid.Size), bytes.Compare)
+	lastRoot := rootsSorted[len(rootsSorted)-1]
+	copy(rng.Start, rootsSorted[0])
+
+	st.Seek(rng, func(k, _ []byte) bool {
+		currRoot := k[len(rng.Prefix) : len(rng.Prefix)+oid.Size]
+		if len(rootsSorted) == 0 || bytes.Compare(currRoot, lastRoot) > 0 {
+			return false
+		}
+		for i := 0; i < len(rootsSorted); i++ {
+			switch bytes.Compare(currRoot, rootsSorted[i]) {
+			case -1:
+				return true
+			case +1:
+				rootsSorted = rootsSorted[i+1:]
+				i = -1
+				continue
+			case 0:
+				childrenWithParentID = append(childrenWithParentID, slices.Clone(k[len(rng.Prefix)+oid.Size+1:]))
+				rootsSorted = rootsSorted[i+1:]
+				return true
+			}
+		}
+
+		return true
```

Comment:
seems like here we are done

Reply:
if move empty check, yes, changed

Reply:
don't we always exit the loop with an empty slice? I think we can just `return false` after `for`